src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -446,3 +467,32 @@ LocalTableEmpty(Oid tableId)
 
 	return localTableEmpty;
 }
+
+
+/* CreateTruncateTrigger creates a truncate trigger on table identified by relationId
+ * and assigns citus_truncate_trigger() as handler. The new trigger is named as
+ * citus_truncate_trigger_on_ + schemaName.tableName. Trigger name for relation my_table
+ * from schema my_schema will be citus_truncate_trigger_on_my_schema.my_table to prevent
+ * name conflicts.
+ */
+static void
+CreateTruncateTrigger(Oid relationId, char *qualifiedRelationName)
+{
+	CreateTrigStmt *trigger = NULL;
+	StringInfo triggerName = makeStringInfo();
+	appendStringInfo(triggerName, "citus_truncate_trigger_on_%s", qualifiedRelationName);
```

Comment:
postgresql assigns autogenerated suffix when trigger is created in **internal** mode. giving and explicit name to trigger is not necessary anymore